lib/jelix/core/request/jClassicRequest.class.php

```diff
@@ -25,18 +25,23 @@ class jClassicRequest extends jRequest {
     protected function _initParams(){
 
         $url  = jUrl::getEngine()->parseFromRequest($this, $_GET);
+        $this->params = $url->params;
 
         if ($_SERVER['REQUEST_METHOD'] == 'PUT') {
             $_PUT = $this->readHttpBody();
             if (is_string($_PUT)) {
-                $this->params = $url->params;
                 $this->params['__httpbody'] = $_PUT;
             } else {
-                $this->params = array_merge($url->params, $_PUT);
+                $this->params = array_merge($this->params, $_PUT);
             }
         }
-        else {
-            $this->params = array_merge($url->params, $_POST);
+        elseif($_SERVER['REQUEST_METHOD'] == 'POST') {
+            $data = $this->readHttpBody();
```

Comment:
I will improve this code to not do things twice, sure.

In the current 1.6.x code, I doubt that body is parsed for requests other than POST (OPTIONS, DELETE).
I have found some docs saying `$_POST` in PHP is badly named and should be populated whenever there's a body with content type like "application/x-www-form-urlencoded" or "multipart/form-data", but it's not.
I made a simple script with `var_dump($_POST);` and made requests on it with different methods and content-types. Only POST requests with one of the two content-types above are automatically parsed into `$_POST`, as PHP documentation says too.

So I do see 3 cases:
- GET : do not parse body
- `$_SERVER['REQUEST_METHOD'] == 'POST' && isset($_SERVER['CONTENT_TYPE']) && ($_SERVER['CONTENT_TYPE'] == 'application/x-www-form-urlencoded' || $_SERVER['CONTENT_TYPE'] == 'multipart/form-data')`: merge with $_POST
- anything else: parse body and merge

I'll propose a patch based on this.